framerail/src/routes/[slug]/[...extra]/page.svelte

```diff
@@ -789,6 +1006,428 @@
+          <div class="revision-attribute revision-type">
+            {$page.data.internationalization?.["wiki-page-file-revision-type"]}
+          </div>
+          <div class="revision-attribute created-at">
+            {$page.data.internationalization?.["wiki-page-file.created-at"]}
+          </div>
+          <div class="revision-attribute user">
+            {$page.data.internationalization?.["wiki-page-revision-user"]}
+          </div>
+          <div class="revision-attribute page">
+            {$page.data.internationalization?.["wiki-page-file.page"]}
+          </div>
+          <div class="revision-attribute name">
+            {$page.data.internationalization?.["wiki-page-file.name"]}
+          </div>
+          <div class="revision-attribute mime">
+            {$page.data.internationalization?.["wiki-page-file.mime"]}
+          </div>
+          <div class="revision-attribute size">
+            {$page.data.internationalization?.["wiki-page-file.size"]}
+          </div>
+          <div class="revision-attribute comments">
+            {$page.data.internationalization?.["wiki-page-revision-comments"]}
+          </div>
+        </div>
+        <!-- Here we sort the list in descending order. -->
+        {#each [...fileRevisionMap].sort((a, b) => b[0] - a[0]) as [_, revisionItem] (revisionItem.revision_number)}
+          <div class="revision-row" data-id={revisionItem.revision_id}>
+            <div class="revision-attribute action">
+              {#if ["create", "regular"].includes(revisionItem.revision_type)}
+                <button
+                  class="action-button revision-rollback clickable"
+                  type="button"
+                  on:click|stopPropagation={() => {
+                    fileEditId = revisionItem.file_id
+                    rollbackFileRevision(revisionItem.revision_number)
+                  }}
+                >
+                  {$page.data.internationalization?.["wiki-page-revision-rollback"]}
+                </button>
+              {/if}
+            </div>
+            <div class="revision-attribute revision-number">
+              {revisionItem.revision_number}
+            </div>
+            <div class="revision-attribute revision-type">
+              {$page.data.internationalization?.[
+                `wiki-page-file-revision-type.${revisionItem.revision_type}`
+              ]}
+            </div>
+            <div class="revision-attribute created-at">
+              {new Date(revisionItem.created_at).toLocaleString()}
+            </div>
+            <div class="revision-attribute user">
+              {revisionItem.user_id}
+            </div>
+            <div class="revision-attribute page">
+              {revisionItem.page_id}
+            </div>
+            <div class="revision-attribute name">
+              {revisionItem.name}
+            </div>
+            <div class="revision-attribute mime">
+              {revisionItem.mime_hint.split(";")[0]}
```

Comment:
I don't think MIMEs will always have a semicolon, and also, why can't we just show the whole MIME type to users?

Reply:
I personally think the charset is kind of an unnecessary info, but we can go with showing the whole thing

Reply:
My thinking is that people interested in MIME types are likely to be technical, and so then providing the whole string is more useful than not.

Reply:
I am pretty sure there is a way to get a textual, more human-readable file type from a MIME. However I don't know if our current magic library provides it, nor what its internationalization status is.

Reply:
So I just checked, the current crate we use, `filemagic`, doesn't do this. So adding a "human-readable file description" will need to be an extension added later if that's what we do.

Reply:
Wikidot provides the file type as a pretty rudimentary human readable string, for example it shows .png as "PNG image data" but .js as "No description"